File: solutions/0027_remove-element.py
# Problem: https://leetcode.com/problems/remove-element

class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        removed = 0
        for i in range(len(nums)):
            curr = nums[i - removed]
            if curr == val:
                nums.pop(i - removed)
                removed += 1

File: solutions/0027_remove-element.java
// Problem: https://leetcode.com/problems/remove-element
// Runtime: 0 ms

class Solution {
    public int removeElement(int[] nums, int val) {
        int ptr1 = 0; // next location to place elem
        int removed = 0;

        for (int ptr2 = 0; ptr2 < nums.length; ptr2++) {
            if (nums[ptr2] != val) {
                nums[ptr1] = nums[ptr2];
                ptr1++;
            }
        }

        return ptr1;
    }
}